### PR TITLE
refactor: deploy on Kubernetes

### PR DESCRIPTION
**Pull request by nicecui**
## What's Changed in this PR

* Split the `kubernetes.md` to the `Deploy on Kubernetes` directory.
* Add more content to `Deploy on Kubernetes`.
* There is still a lot of content that needs to be added in the future.

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

## Summary by CodeRabbit

- **Documentation**
  - Updated English user guide with detailed instructions for deploying a GreptimeDB cluster on Kubernetes using Helm or `kubectl`.
  - Revised English overview guide on prerequisites and tools for Kubernetes deployment.
  - Added new Chinese documentation mirroring the English guides for deploying GreptimeDB cluster and using the GreptimeDB Operator.

### CODE REVIEW

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/greptimedb-operator.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+## 安装 GreptimeDB Opertator
+
+通过利用 [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)，
+[GreptimeDB Operator](https://github.com/GreptimeTeam/greptimedb-operator) 可以有效管理 [Kubernetes](https://kubernetes.io/) 上的 GreptimeDB 集群。
+这个 operator 可以抽象出维护高可用 GreptimeDB 集群的模式。
+
+你可以使用 Helm 来安装 GreptimeDB Operator。
+
+```shell
+helm install greptimedb-operator greptime/greptimedb-operator -n greptimedb-admin
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/greptimedb-operator.md:12]: suggestion: Use alicloud docker registry
```
helm upgrade \
  --install \
  --set image.registry=greptime-registry.cn-hangzhou.cr.aliyuncs.com \
  --create-namespace \
  greptimedb-operator greptime/greptimedb-operator \
  -n greptimedb-admin
  ```

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+安装完成后，你可以从安装日志中获取 etcd 集群的 endpoint `etcd.etcd.svc.cluster.local:2379`。
+该 endpoint 在后续步骤部署 GreptimeDB 集群时需要使用。
+
+### 创建 GretpimeDB 集群
+
+部署 GreptimeDB 集群，确保它连接到先前建立的 etcd 集群：
+
+```shell
+helm install greptimedb greptime/greptimedb-cluster \
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:27]: suggestion: Use alicloud docker registry
```
helm upgrade \
  --install greptimedb \
  --set image.registry=image.registry=greptime-registry.cn-hangzhou.cr.aliyuncs.com \
  --set initializer.registry=image.registry=greptime-registry.cn-hangzhou.cr.aliyuncs.com \
  --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
  greptime/greptimedb-cluster \
  -n default
```

File: docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+### Create an etcd cluster
+
+First, establish an etcd cluster to support GreptimeDB by executing the following command:
+
+```shell
+helm install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
+  --set replicaCount=3 \
+  --set auth.rbac.create=false \
+  --set auth.rbac.token.enabled=false \
+  -n etcd
```
Comment (daviderli614) [inline docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:17]: suggestion change:
```
helm install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
  --set replicaCount=3 \
  --set auth.rbac.create=false \
  --set auth.rbac.token.enabled=false \
  --create-namespace \
  -n etcd
```

File: docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+The endpoints are required for deploying the GreptimeDB cluster in the subsequent step.
+
+### Create a GretpimeDB cluster
+
+Deploy the GreptimeDB cluster, ensuring it connects to the previously established etcd cluster:
+  
+```shell
+helm install greptimedb greptime/greptimedb-cluster \
+  --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
+  -n greptimedb-cluster
```
Comment (daviderli614) [inline docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:31]: suggestion change:
```
helm install greptimedb greptime/greptimedb-cluster \
  --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
  --create-namespace \
  -n greptimedb-cluster
```

File: docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+```shell
+helm upgrade \
+      --install etcd \
+      --set replicaCount=3 \
+      --set resources.requests.cpu=100m \
+      --set auth.rbac.create=false \
+      --set auth.rbac.token.enabled=false \
+      oci://registry-1.docker.io/bitnamicharts/etcd \
+      -n etcd
```
Comment (daviderli614) [inline docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:93]: suggestion change:
```
helm upgrade \
      --install etcd \
      --set replicaCount=3 \
      --set resources.requests.cpu=100m \
      --set auth.rbac.create=false \
      --set auth.rbac.token.enabled=false \
      --create-namespace \
      oci://registry-1.docker.io/bitnamicharts/etcd \
      -n etcd
```

File: docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      --install greptimedb \
+      --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
+      --set base.podTemplate.main.resources.requests.cpu=100m \
+      --set base.podTemplate.main.resources.requests.memory=256Mi \
+      --set datanode.podTemplate.main.resources.requests.cpu=100m \
+      --set datanode.podTemplate.main.resources.requests.memory=256Mi \
+      --set frontend.podTemplate.main.resources.requests.cpu=100m \
+      --set frontend.podTemplate.main.resources.requests.memory=256Mi \
+      greptime/greptimedb-cluster \
+      -n greptimedb-cluster
```
Comment (daviderli614) [inline docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:106]: suggestion change:
```
helm upgrade \
      --install greptimedb \
      --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
      --set base.podTemplate.main.resources.requests.cpu=100m \
      --set base.podTemplate.main.resources.requests.memory=256Mi \
      --set datanode.podTemplate.main.resources.requests.cpu=100m \
      --set datanode.podTemplate.main.resources.requests.memory=256Mi \
      --set frontend.podTemplate.main.resources.requests.cpu=100m \
      --set frontend.podTemplate.main.resources.requests.memory=256Mi \
      greptime/greptimedb-cluster \
      --create-namespace \
      -n greptimedb-cluster 
```

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+首先，通过执行以下命令来建立一个支持 GreptimeDB 的 etcd 集群：
+
+```shell
+helm install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
+  --set replicaCount=3 \
+  --set auth.rbac.create=false \
+  --set auth.rbac.token.enabled=false \
+  -n etcd
+```
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:17]: suggestion change:
```
helm install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
  --set replicaCount=3 \
  --set auth.rbac.create=false \
  --set auth.rbac.token.enabled=false \
  --set image.registry=greptime-registry.cn-hangzhou.cr.aliyuncs.com \
  --set image.tag=3.5.12 \
  --create-namespace \
  -n etcd
```

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+```shell
+helm upgrade \
+      --install etcd \
+      --set replicaCount=3 \
+      --set resources.requests.cpu=100m \
+      --set auth.rbac.create=false \
+      --set auth.rbac.token.enabled=false \
+      oci://registry-1.docker.io/bitnamicharts/etcd \
+      -n etcd
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:91]: suggestion change:
```
helm upgrade \
      --install etcd \
      --create-namespace \
      --set replicaCount=3 \
      --set resources.requests.cpu=100m \
      --set auth.rbac.create=false \
      --set auth.rbac.token.enabled=false \
      oci://registry-1.docker.io/bitnamicharts/etcd \
      -n etcd
```

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+      --install greptimedb \
+      --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
+      --set base.podTemplate.main.resources.requests.cpu=100m \
+      --set base.podTemplate.main.resources.requests.memory=256Mi \
+      --set datanode.podTemplate.main.resources.requests.cpu=100m \
+      --set datanode.podTemplate.main.resources.requests.memory=256Mi \
+      --set frontend.podTemplate.main.resources.requests.cpu=100m \
+      --set frontend.podTemplate.main.resources.requests.memory=256Mi \
+      greptime/greptimedb-cluster \
+      -n greptimedb-cluster
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:104]: suggestion change:
```
helm upgrade \
      --install greptimedb \
      --create-namespace \
      --set meta.etcdEndpoints=etcd.etcd.svc.cluster.local:2379 \
      --set base.podTemplate.main.resources.requests.cpu=100m \
      --set base.podTemplate.main.resources.requests.memory=256Mi \
      --set datanode.podTemplate.main.resources.requests.cpu=100m \
      --set datanode.podTemplate.main.resources.requests.memory=256Mi \
      --set frontend.podTemplate.main.resources.requests.cpu=100m \
      --set frontend.podTemplate.main.resources.requests.memory=256Mi \
      greptime/greptimedb-cluster \
      -n greptimedb-cluster 
```

**Review by killme2008 — APPROVED** [review]
LGTM

File: docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+  --set meta.podTemplate.main.resources.requests.cpu=<cpu-resource> \
+  --set meta.podTemplate.main.resources.requests.memory=<mem-resource> \
+  --set datanode.podTemplate.main.resources.requests.cpu=<cpu-resource> \
+  --set datanode.podTemplate.main.resources.requests.memory=<mem-resource> \
+  --set frontend.podTemplate.main.resources.requests.cpu=<cpu-resource> \
+  --set frontend.podTemplate.main.resources.requests.memory=<mem-resource> \
+  --create-namespace \
+  -n greptimedb-cluster
+```
+
```
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:54]: It's better to give a setup example to copy and run with the smallest resource limits.

Comment (killme2008) [reply docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:54]: Giving a link to the capacity plan would be good.

Comment (nicecui) [reply docs/nightly/en/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:54]: The details of capacity plan for GreptimeDB Cluster will be written in the next PR.

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/overview.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+kubectl create namespace etcd
+kubectl create namespace greptimedb-cluster
+```
+
+### 添加 Helm 仓库
+
+使用以下命令添加GreptimeDB Helm chart仓库。
+
+```shell
+helm repo add greptime https://greptimeteam.github.io/helm-charts/
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/overview.md:44]: suggestion change:
```
helm repo add greptime https://downloads.greptime.cn/releases/charts/package
```

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+```shell
+helm upgrade \
+  --install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
+  --set replicaCount=3 \
+  --set auth.rbac.create=false \
+  --set auth.rbac.token.enabled=false \
+  --set image.registry=greptime-registry.cn-hangzhou.cr.aliyuncs.com \
+  --set image.tag=3.5.12 \
+  --create-namespace \
+  -n etcd
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:20]: suggestion change:
```
helm upgrade \
  --install etcd greptime/etcd \
  --set replicaCount=3 \
  --set auth.rbac.create=false \
  --set auth.rbac.token.enabled=false \
  --set image.registry=greptime-registry.cn-hangzhou.cr.aliyuncs.com \
  --set image.tag=3.5.12 \
  --create-namespace \
  -n etcd
```

File: docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+  -n greptimedb-cluster
+```
+
+### 设置资源
+
+GreptimeDB Helm charts 能够为部署中的每个组件指定资源请求和限制。
+以下是如何配置这些设置的示例：
+
+```shell
+helm install greptimedb greptime/greptimedb-cluster \
```
Comment (daviderli614) [inline docs/nightly/zh/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:60]: suggestion change: 
`helm install greptimedb greptimedb-cluster \`

**Review by daviderli614 — APPROVED** [review]

**Pull request merged**